cmd/kueuectl/app/list/list_pods.go

```diff
@@ -0,0 +1,245 @@
+	addLabelSelectorFlagVar(cmd, &o.LabelSelector)
+	addForObjectFilterFlagVar(cmd, &o.UserSpecifiedForObject)
+
+	return cmd
+}
+
+// Complete takes the command arguments and infers any remaining options.
+func (o *PodOptions) Complete(clientGetter util.ClientGetter) error {
+	var err error
+
+	o.Namespace, _, err = clientGetter.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.Clientset, err = clientGetter.K8sClientSet()
+	if err != nil {
+		return err
+	}
+
+	if !o.PrintFlags.OutputFlagSpecified() {
+		o.PrintObj = printPodTable
+	} else {
+		printer, err := o.PrintFlags.ToPrinter()
+		if err != nil {
+			return err
+		}
+		o.PrintObj = printer.PrintObj
+	}
+
+	o.ForObject, err = parseForObjectFilterFlag(o.UserSpecifiedForObject)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run prints the pods for a specific Job
+func (o *PodOptions) Run(ctx context.Context) error {
+	err := o.listPods(ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// listPods lists the pods based on the given --for object
+func (o *PodOptions) listPods(ctx context.Context) error {
+	namespace := o.Namespace
+	if o.AllNamespaces {
+		namespace = ""
+	}
+
+	continueToken := ""
+	filteredPodListSlice := make([]*corev1.PodList, 0)
+	for {
+		podList, err := o.Clientset.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+			LabelSelector: o.LabelSelector,
+			FieldSelector: o.FieldSelector,
+			Continue:      continueToken,
+		})
+		if err != nil {
+			return err
+		}
+
+		if fp := o.filterPodsByOwnerRef(podList); len(fp) != 0 {
+			podList.Items = fp
+			filteredPodListSlice = append(filteredPodListSlice, podList)
+		}
+
+		if podList.Continue == "" {
+			break
+		}
+		continueToken = podList.Continue
+	}
+
+	// handle if no filtered podList found
+	if len(filteredPodListSlice) == 0 {
+		if !o.AllNamespaces {
+			fmt.Fprintf(o.ErrOut, "No resources found in %s namespace.\n", o.Namespace)
+		} else {
+			fmt.Fprintln(o.ErrOut, "No resources found.")
+		}
+		return nil
+	}
+
+	// now range over the filteredPodListSlice and print them
+	for _, filteredPodList := range filteredPodListSlice {
+		if err := o.PrintObj(filteredPodList, o.Out); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *PodOptions) filterPodsByOwnerRef(podList *corev1.PodList) []corev1.Pod {
+	filteredPods := make([]corev1.Pod, 0, len(podList.Items))
+
+	for i := range podList.Items {
+		for _, ownerRef := range podList.Items[i].OwnerReferences {
+			gv, _ := schema.ParseGroupVersion(ownerRef.APIVersion)
+
+			if strings.EqualFold(o.ForObject.Kind, ownerRef.Kind) && strings.EqualFold(o.ForObject.Name, ownerRef.Name) {
+				if o.ForObject.APIGroup == "" || strings.EqualFold(o.ForObject.APIGroup, gv.Group) {
+					filteredPods = append(filteredPods, podList.Items[i])
+					break
+				}
+			}
+		}
+	}
+
+	return filteredPods
+}
+
+// printPodTable is a printer function for PodList objects.
+var _ printers.ResourcePrinterFunc = printPodTable
```

Comment:
Could you please move it to separate printer? It will allow us to use extra options for tablePrinter. Like [here](https://github.com/kubernetes-sigs/kueue/blob/4bf04036d7876cfef72b9a4e5be398c9d47a14fd/cmd/kueuectl/app/list/list_localqueue_printer.go).

Reply:
Done https://github.com/kubernetes-sigs/kueue/pull/2280/commits/ea96ded0ce3c4292a0a9b45632792cbd09f7f706